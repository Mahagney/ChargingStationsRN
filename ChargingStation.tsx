import { StyleSheet, Text, View } from "react-native";
import Logo from "./assets/RightArrow.svg"
import Star from "./assets/star.svg";

const calculateDistance = (distance: number) => {
    return distance>1000 ? `${distance/1000}km` : `${distance}m`;
}

export type ChargingPointType = {
    "availableChargePoints": string, 
    "distance": string, 
    "isFavourite": boolean, 
    "location": string, 
    "stationName": string, 
    "totalChargePoints": string
}

export const ChargingStation = ({availableChargePoints, totalChargePoints, stationName, location, isFavourite, distance, isLast, isFirst}:ChargingPointType & {isLast:boolean, isFirst: boolean}) => {
    
    const style = StyleSheet.create({
        container: {
            display: 'flex',
            flexDirection: 'row',
            width: '100%',
            height: 70,
            borderBottomWidth: isLast ? 0 : 1,
            borderBottomColor: '#b8bfc4',
        },
        distanceContainer: {
            display: 'flex',
            flexDirection: 'column',
            alignItems:'center',
            backgroundColor: '#e6ebee',
            padding: 16,
            overflow: 'hidden',
            width: 80,
            zIndex: 2,
            borderTopRightRadius: isFirst ? 10 : 0,
            borderBottomRightRadius: isLast ? 10 : 0
        },
    });
    return (
        <View style={style.container}>
            <View style={styles.availableContainer}>
                <Text style={styles.totalText}> {availableChargePoints}/{totalChargePoints} </Text>
                <Text> Available </Text>
            </View>
            <View style={styles.addressContainer}>
                <View style={styles.starContainer}>
                    <Text style={ styles.addressText}>{ stationName.length >15 ? `${stationName.slice(0,15)}...` : stationName } </Text>
                    {isFavourite && <Star style={styles.star}/>}
                </View>
                <Text style={ styles.addressText}>{ location.length >18 ? location.slice(0,18) : location }</Text>
            </View>
            <View style={style.distanceContainer}>
                <Logo/>
                <Text>{calculateDistance(parseInt(distance))}</Text>    
            </View>            
        </View>
    );
}

const styles = StyleSheet.create({
    availableContainer: {
        display: 'flex',
        flexDirection: 'column',
        padding: 16,
    },
    starContainer: {
        display: 'flex',
        flexDirection:'row'
    },
    star: {
        marginTop: 4
    },
    addressContainer: {
        display: 'flex',
        flexDirection: 'column',
        padding: 16,
        flexGrow: 1
    },
    totalText: {
        fontWeight: '700',
        fontSize: 18,
    },
    addressText: {
        fontSize: 12,
        color: '#383d3f',
    }
});