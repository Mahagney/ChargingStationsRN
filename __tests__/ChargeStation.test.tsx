import {create} from 'react-test-renderer';
import {ChargingStation} from '../components/molecules/ChargingStation.tsx';
import { ChargingPointType } from '../components/molecules/ChargingStation.tsx';
import {expect, it} from '@jest/globals';

jest.mock('../assets/star.svg', () => () => null) 
jest.mock('../assets/RightArrow.svg', () => () => null) 

const data: ChargingPointType[] = [
    {"availableChargePoints": "5", "distance": "780", "isFavourite": true, "location": "LE11 EC3, UK", "stationName": "Loughborough towncentre hall2", "totalChargePoints": "5"}, 
{ "availableChargePoints": "9", "distance": "140", "isFavourite": true, "location": "W2 5TY, UK", "stationName": "Paddington 36", "totalChargePoints": "18"}, 
{ "availableChargePoints": "3", "distance": "12000", "isFavourite": false, "location": "W2 5TY, UK", "stationName": "London 2", "totalChargePoints": "10"}, 
{"availableChargePoints": "4", "distance": "12000", "isFavourite": false, "location": "W2 5TY, UK", "stationName": "London 4", "totalChargePoints": "10"}, 
{"availableChargePoints": "5", "distance": "12000", "isFavourite": false, "location": "W2 5TY, UK", "stationName": "London 5", "totalChargePoints": "10"}, 
{"availableChargePoints": "8", "distance": "12000", "isFavourite": false, "location": "W2 5TY, UK", "stationName": "London 6", "totalChargePoints": "10"}, 
{ "availableChargePoints": "1", "distance": "12000", "isFavourite": false, "location": "W2 5TY, UK", "stationName": "London 8", "totalChargePoints": "10"}];
// render the component

it('renders correctly', () => {
  const tree = create(<ChargingStation isLast={false} isFirst={false} {...data[0]} />).toJSON();
  expect(tree).toMatchSnapshot();
});