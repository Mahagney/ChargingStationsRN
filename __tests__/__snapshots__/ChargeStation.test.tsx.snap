// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "borderBottomColor": "#b8bfc4",
      "borderBottomWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "height": 70,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "display": "flex",
        "flexDirection": "column",
        "padding": 16,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "700",
        }
      }
    >
       
      5
      /
      5
       
    </Text>
    <Text>
       Available 
    </Text>
  </View>
  <View
    style={
      {
        "display": "flex",
        "flexDirection": "column",
        "flexGrow": 1,
        "padding": 16,
      }
    }
  >
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "color": "#383d3f",
            "fontSize": 12,
          }
        }
      >
        Loughborough to...
         
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#383d3f",
          "fontSize": 12,
        }
      }
    >
      LE11 EC3, UK
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#e6ebee",
        "borderBottomRightRadius": 0,
        "borderTopRightRadius": 0,
        "display": "flex",
        "flexDirection": "column",
        "overflow": "hidden",
        "padding": 16,
        "width": 80,
        "zIndex": 2,
      }
    }
  >
    <Text>
      780m
    </Text>
  </View>
</View>
`;
